cvpy/tests/test_credentials.py

```diff
@@ -0,0 +1,73 @@
+import sys
+import unittest
+from pathlib import PosixPath
+
+import xmlrunner
+
+from cvpy.annotation.base.Credentials import Credentials
+
+
+class TestCredentials(unittest.TestCase):
+
+    def test_credentials_user_password(self):
+        user = 'test_user'
+        password = 'test_password'
+
+        credentials = Credentials(user=user, password=password)
+
+        self.assertEqual(credentials.user, user)
+        self.assertEqual(credentials.password, password)
+        self.assertIsNone(credentials.token)
+
+    def test_credentials_with_token(self):
+        token = 'abc1def2ghi'
+        credentials = Credentials(token=token)
+
+        self.assertIsNotNone(credentials.token)
+        self.assertEqual(credentials.get_auth_header(), dict(Authorization=f'token {token}'))
+
+    def test_credentials_token_not_set(self):
+        try:
+            user = 'test_user'
+            password = 'test_password'
+
+            credentials = Credentials(user=user, password=password)
+            credentials.get_auth_header()
+        except Exception as e:
+            self.assertEqual(str(e), 'Token is not set.')
+
+    def test_credentials_default_authfile(self):
+        default_auth_file_path = PosixPath('~/.annotation_auth').expanduser()
+        self.assertTrue(default_auth_file_path.is_file(), 'The default file ~/.annotation_auth does not exist.')
+
+        credentials = Credentials()
+
+        self.assertTrue(credentials.token or (credentials.user and credentials.password))
+
+    def test_credentials_authfile_with_token(self):
+        auth_file_path = PosixPath(TestCredentials.auth_file_with_token)
+        self.assertTrue(auth_file_path.is_file(),
+                        f'The file {TestCredentials.auth_file_with_token} does not exist.')
+
+        credentials = Credentials(auth_file=TestCredentials.auth_file_with_token)
+        self.assertIsNotNone(credentials.token)
+
+    def test_credentials_authfile_with_user_password(self):
+        auth_file_path = PosixPath(TestCredentials.auth_file_with_user_password)
+        self.assertTrue(auth_file_path.is_file(),
+                        f'The file {TestCredentials.auth_file_with_user_password} does not exist.')
```

Comment:
Can we have a brief comment right before the definitions of each of these tests describing what it's actually testing? This will be similar to how we comment unit tests in C: https://grok.sas.com/xref/mva-vbviya/tkcv/test/tkimcore/src/imageSobel.c#15

Reply:
Done